### ISSUE TITLE
Caller is not running foreground, code = -100

### ISSUE THREAD

**Issue by MarsupiL** [NiklasMerz/cordova-plugin-fingerprint-aio#340]
## Description

* FaceId scan fails on iOS in our production environment.
* Fingerprint.show is rejected with error code -100 and error message "Caller is not running foreground"

I am using the Capacitor App state to detect when the app is in foreground on app resume and call our Biometrics service. Fingerprint.isAvailable succeeds and returns "face" but Fingerprint.show fails with the error -100 mentioned above.  
The same code used to work fine until February 8th 2021. 

## Environment

* Plugin version: 4.0.2
* Running on Capacitor 2.4.7
* iOS 14.0 -> 14.4

**Comment by epiphanizer** [comment]
I'm seeing this as well. Were you able to find a solution here?

**Comment by eamondunne** [comment]
I recently came across this exact error on iOS. In my case, it was happening after the app received a push notification to a locked device. 
After some testing I found that if the call happened directly after unlocking the device it would fail with this error. Adding a delay of a few seconds resolved the issue. This would suggest to me at a native level perhaps the biometrics isn't fully ready on the show() call, or is being blocked by the screen unlock, however that is entirely speculative so take it with a pinch of salt.

Our current workaround is to catch the error then wait a few seconds and try again (asynchronously). Not ideal but it did resolve the problem.

Hope this helps in some way for anyone else working on this issue.
